src/helper/helper.js

```diff
@@ -48,165 +48,163 @@ goog.module('helper');
 
 const {type, hasOwn, isPlainObject} = goog.require('plain');
 
-
 /**
- * Creates a new helper object for the given dataLayer.
+ * A helper that will listen for new messages on the given dataLayer.
+ * Each new message will be merged into the helper's "abstract data model".
+ * This internal model object holds the most recent value for all keys which
+ * have been set on messages processed by the helper.
  *
- * @constructor
- * @param {!Array.<!Object>} dataLayer The dataLayer to help with.
- * @param {function(!Object, !Object)=} opt_listener The callback function to
- *     execute when a new state gets pushed onto the dataLayer.
- * @param {boolean=} opt_listenToPast If true, the given listener will be
- *     executed for state changes that have already happened.
+ * You can retrieve values from the data model by using the helper's get method.
  */
-const DataLayerHelper = function(dataLayer, opt_listener, opt_listenToPast) {
-
+class DataLayerHelper {
   /**
-   * The dataLayer to help with.
-   * @type {!Array.<!Object>}
-   * @private
-   */
-  this.dataLayer_ = dataLayer;
-
-  /**
-   * The listener to notify of changes to the dataLayer.
-   * @type {function(!Object, !Object)}
-   * @private
+   * Creates a new helper object for the given dataLayer.
+   *
+   * @param {!Array.<!Object>} dataLayer The dataLayer to help with.
+   * @param {function(!Object, !Object)=} opt_listener The callback function to
+   *     execute when a new state gets pushed onto the dataLayer.
+   * @param {boolean=} opt_listenToPast If true, the given listener will be
+   *     executed for state changes that have already happened.
    */
-  this.listener_ = opt_listener || function() {};
-
-  /**
-   * The internal marker for checking if the listener is currently on the stack.
-   * @type {boolean}
-   * @private
-   */
-  this.executingListener_ = false;
+  constructor(dataLayer, opt_listener, opt_listenToPast) {
+    /**
+     * The dataLayer to help with.
+     * @type {!Array.<!Object>}
```

Comment:
We can probably make a lot of these members "@const":
https://google.github.io/styleguide/jsguide.html#features-classes-fields

Also, we can simplify these by removing "@type" if the member is one of @private, @const, or @protected. Again, ok with me if you plan on doing this in another PR.

Reply:
I changed these over to @const but left the type annotations. Do you suggest removing them as they are optional?